src/realm/path.hpp

```diff
@@ -256,6 +256,11 @@ class ExtendedColumnKey {
     ObjKey get_link_target(const Obj& obj) const;
     Mixed get_value(const Obj& obj) const;
 
+    // get String ID for the obj, it makes sense to call this method only if the col_key type is either Mixed or
+    // String.
+    using StringID = size_t;
```

Comment:
I don't like to repeat this, but also I don't like to have a global alias to call `size_t` `StringID`

Reply:
Maybe make it global, but under a `realm::compression` namespace? It can be confusing to have multiple definitions too...